### PR TITLE
feat: The first version of the scanner

### PR DESCRIPTION
**Pull request by titusjaka**

### CODE REVIEW

File: SPECIFICATION.md [hunk 1]
```diff
@@ -69,10 +69,16 @@ value_with_double_quotes="A value between double quotation marks."
 
 - If a value is empty, it's interpreted as an empty string ''.
 - If a line lacks `=` character, the value is interpreted as an empty string ''.
+- If a file contains 2 or more equal `<name>`-s, then the last value is used.
```
Comment (i-sevostyanov) [inline SPECIFICATION.md]:
```suggestion
- If a file contains 2 or more identical `<name>`-s, then the last value is used.
```

**Review by i-sevostyanov — APPROVED** [review]

File: SPECIFICATION.md (hunk 1 above)
Comment (nonameffh) [inline SPECIFICATION.md]: May be it will be useful to specify how multiple files will be processed

File: scanner/scanner.go [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+type Scanner struct {
+	input      string
+	ch         rune // current character
+	prevOffset int  // position before current character
+	offset     int  // character offset
+	peekOffset int  // position after current character
+}
+
+// New returns new Scanner.
+func New(input string) *Scanner {
```
Comment (nonameffh) [inline scanner/scanner.go:26]: It might be helpful to give an example of 'Scanner' usage

**Review by nonameffh — APPROVED** [review]

File: scanner/scanner.go (hunk 2 above)
Comment (titusjaka) [reply scanner/scanner.go:26]: I guess there is no way anyone will use `Scanner` without `Parser` 😅 Thus, there is no actual need for an example for it.

File: SPECIFICATION.md (hunk 1 above)
Comment (titusjaka) [reply SPECIFICATION.md]: Yep, sure. But only after we add support for multiple files. I guess the first version of the godenv will lack this feature.

File: scanner/scanner.go (hunk 2 above)
Comment (nonameffh) [reply scanner/scanner.go:26]: Ok, this is a little bit of my paranoia, everything that is exported can be used against you.

**Pull request merged**